File: KnowledgeCenterServer/_Green/KnowledgeCenter.Green.Contracts/Publication.cs
using System;

namespace KnowledgeCenter.Green.Contracts
{
    public class Publication
    {
        public int Id { get; set; }
        public string Message { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }
        public DateTime PublicationDate { get; set; }
        public bool IsPublished { get; set; }
        public int PublicationTypeId { get; set; }
        public PublicationType PublicationType { get; set; }
        public int UserId {get; set;}
        public User User { get; set; }
    }
}

File: KnowledgeCenterClient/src/app/green/models/Publication.ts
import { User } from 'src/app/shared/models/User';
import { SimpleEnum } from 'src/app/shared/models/SimpleEnum';

export interface Publication {
    id: number;
    message: string;
    creationDate: Date;
    modificationDate: Date;
    publicationDate: Date;
    isPublished: boolean;
    publicationTypeId: number;
    publicationType: SimpleEnum;
    user: User;
}
